assembly/src/parsers/ast/serde/deserialization.rs

```diff
@@ -67,15 +67,26 @@ impl<'a> ByteReader<'a> {
     }
 
     pub fn read_string(&mut self) -> Result<String, SerializationError> {
-        self.check_eor(1)?;
-        let length = self.bytes[self.pos];
-        self.pos += 1;
+        let length = self.read_u8()?;
```

Comment:
Let's change the name of this function to be: `read_proc_name()`. And also, let's do the same serialization: `write_string()` -> `write_proc_name()`.